frontend/class-frontend.php

```diff
@@ -1521,18 +1521,7 @@ public function clean_permalink() {
 		global $wp_query;
 
 		// Recreate current URL.
-		$cururl = 'http';
-		if ( isset( $_SERVER['HTTPS'] ) && $_SERVER['HTTPS'] == 'on' ) {
-			$cururl .= 's';
-		}
-		$cururl .= '://';
-
-		if ( $_SERVER['SERVER_PORT'] != '80' && $_SERVER['SERVER_PORT'] != '443' ) {
-			$cururl .= $_SERVER['SERVER_NAME'] . ':' . $_SERVER['SERVER_PORT'] . $_SERVER['REQUEST_URI'];
-		}
-		else {
-			$cururl .= $_SERVER['SERVER_NAME'] . $_SERVER['REQUEST_URI'];
-		}
+		$cururl = home_url( sanitize_text_field( $_SERVER['REQUEST_URI'] ) );
```

Comment:
I would choose this way:

`$cururl = home_url( sanitize_text_field( filter_input( INPUT_SERVER, 'REQUEST_URI' ) ) );`